### DISCUSSION TITLE
Does it matter if I import React or not?

### DISCUSSION THREAD

**Discussion by lloydjatkinson** [vercel/next.js#14896] [Help] [answered]
I use ESLint and the default index.js file is causing a warning around JSX and React - `'React' must be in scope when using JSX    react/react-in-jsx-scope`.

I can either `import React from 'react'` or turn this rule off. I've imported it for now - but will this have some weird problem with how Next statically builds the app?

**Comment by Timer** [comment, marked as answer]
Next.js automatically imports React for you in the most optimized way. You can safely turn off this lint rule.

If you're going to import yourself, you should use the following for the most optimized bundle output:

```tsx
import * as React from 'react';
```
